pkg/services/signingkeys/signingkeysimpl/service.go

```diff
@@ -0,0 +1,89 @@
+package signingkeysimpl
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+
+	"github.com/go-jose/go-jose/v3"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/signingkeys"
+)
+
+const (
+	serverPrivateKeyID = "default"
+)
+
+var _ signingkeys.Service = new(Service)
+
+func ProvideEmbeddedSigningKeysService() *Service {
+	s := &Service{
+		log: log.New("auth.key_service"),
+	}
```

Comment:
I'd say to probably add a feature toggle gate here and on the methods

Reply:
As we discussed, the feature flag is not required here, because this service is not completely tied to the new extended auth jwt service + oauth server. However, because the service can change in the near future I added comments to make it clear that it is still under active development.